build/net/src/lib.rs

```diff
@@ -82,23 +98,129 @@ pub fn load_net_config() -> Result<NetConfig> {
     Ok(cfg)
 }
 
-pub fn generate_vlan_consts(
+pub fn generate_port_consts(
     config: &NetConfig,
     mut out: impl std::io::Write,
 ) -> Result<(), std::io::Error> {
-    let vlan = config.vlan.unwrap();
-    let end = vlan.start + vlan.count;
-    if end > 0xFFF {
-        panic!("Invalid VLAN range (must be < 4096)");
+    let ports = config
+        .vlans
+        .values()
+        .map(|v| v.port)
+        .collect::<BTreeSet<_>>()
+        .len();
```

Comment:
it took me a sec to realize that the reason we did this instead of using `Iterator::count()` was that we wanted to count unique ports, which...is obvious in hindsight, so maybe it's not really worth adding a comment on, i dunno...

Reply:
(minor allocation golf that almost certainly doesn't matter: it might be nicer to use a `HashSet` here, since we don't care about order, just uniqueness, and we won't have to allocate for each entry?)

Reply:
lol Cliff literally suggested switching from `HashSet` to `BTreeSet` in [this comment](https://github.com/oxidecomputer/hubris/pull/1835#discussion_r1699203911)

Reply:
welp, disregard that.